### ISSUE TITLE
Implement vstack/hstack and tensor concatenation

### ISSUE THREAD

**Issue by romeric** [romeric/Fastor#103]

**Comment by dorpxam** [comment]
I'm working on an experimental deep neural network based on a tensorflow model and possibly want to use fastor as heart tensor of the network. I don't know if it's really possible and I made my own tiny library from scratch (very preliminary). In the same idea of vstack/hstack & concatenation, there's maybe some usefull similar function to implement (from tensorflow): tf.concat / tf.reshape / tf.squeeze / tf.tile.

**Comment by romeric** [comment]
Yes, all these features get requested quite a lot. I will implement them as soon as I get some free time. Unfortunately these days my hands are pretty tied with other work related projects.

At the end you have to way in how much of your neural network algorithm can be implemented with compile time types. If the answer to that question is "most of them" then I am pretty sure you can use Fastor for it. Also feel free to modify Fastor to make it your own if that suites your workflow better.

Roman
